app/views/shared/_post_errors.html.erb

```diff
@@ -0,0 +1,8 @@
+<% if post.errors.any? %>
+  The form contains <%= pluralize(post.errors.count, "error") %>.
+  <ul>
+    <% post.errors.full_messages.each do |msg| %>
+      <li><%= msg %></li>
+    <% end %>
+  </ul>
+<% end %>
```

Comment:
What about just having an `_errors.html.erb` partial that took a `resource` local variable?  So you'd invoke it like `render partial: "shared/errors", resource: post` or `render partial: "shared/errors", resource: comment`.